File: typescript/src/modules/greeter/GreeterModule.ts
import type { IModule } from '../../core/IModule';

/**
 * Greeter Module — Module thứ 2 trong vũ trụ.
 * Chứng minh thêm module mới = thêm 1 file + 1 dòng register.
 */
export class GreeterModule implements IModule {
  readonly name = 'greeter';
  readonly description = 'Greeting messages (hello, goodbye)';
  readonly commands = ['hello', 'goodbye'] as const;

  execute(command: string, args: string[]): string {
    const name = args.length > 0 ? args.join(' ') : 'World';

    switch (command.toLowerCase()) {
      case 'hello':   return `👋 Hello, ${name}! Welcome to the Universe!`;
      case 'goodbye': return `🌙 Goodbye, ${name}! See you among the stars!`;
      default: return `Unknown command '${command}'. Available: ${this.commands.join(', ')}`;
    }
  }
}

File: python/modules/greeter/greeter_module.py
"""
Greeter Module — Module thứ 2 trong vũ trụ.
Chứng minh thêm module mới = thêm 1 file + 1 dòng register.
"""
from core.module import IModule


class GreeterModule(IModule):

    @property
    def name(self) -> str:
        return "greeter"

    @property
    def description(self) -> str:
        return "Greeting messages (hello, goodbye)"

    @property
    def commands(self) -> list[str]:
        return ["hello", "goodbye"]

    def execute(self, command: str, args: list[str]) -> str:
        name = " ".join(args) if args else "World"

        match command.lower():
            case "hello":   return f"👋 Hello, {name}! Welcome to the Universe!"
            case "goodbye": return f"🌙 Goodbye, {name}! See you among the stars!"
            case _:         return f"Unknown command '{command}'. Available: {', '.join(self.commands)}"
